v3/src/components/graph/hooks/graph-hooks.ts

```diff
@@ -136,15 +137,65 @@ export const usePlotResponders = (props: IPlotResponderProps) => {
     const disposer = dataConfiguration.onAction(action => {
       if (['addCases', 'removeCases', 'setAttributeType'].includes(action.name)) {
         matchCirclesToData({
+          dataset,
           caseIDs: dataConfiguration.cases,
           pointRadius: graphModel.getPointRadius(),
-          dotsElement: dotsRef.current ,
+          dotsElement: dotsRef.current,
           enableAnimation, instanceId
         })
         callRefreshPointPositions(false)
       }
     })
     return () => disposer()
-  }, [enableAnimation, graphModel, callRefreshPointPositions, dotsRef, instanceId])
+  }, [dataset, enableAnimation, graphModel, callRefreshPointPositions, dotsRef, instanceId])
 
 }
+
+const dataTip = d3tip().attr('class', 'graph-d3-tip')/*.attr('opacity', 0.8)*/
+  .attr('data-testid', 'graph-point-data-tip')
+  .html((d: string) => {
+    return "<p>" + d + "</p>"
+  })
+
+export const useDataTips = (dotsRef: React.RefObject<SVGSVGElement>,
+                            dataset: IDataSet | undefined, graphModel: IGraphModel) => {
```

Comment:
This module is getting pretty long. Seems like `useDataTips` merits its own `use-data-tips.ts` module.

Reply:
Done